packages/html-reporter/src/reportView.tsx

```diff
@@ -39,54 +39,77 @@ declare global {
 const testFilesRoutePredicate = (params: URLSearchParams) => !params.has('testId');
 const testCaseRoutePredicate = (params: URLSearchParams) => params.has('testId');
 
+type TestModelSummary = {
+  files: TestFileSummary[];
+  tests: TestCaseSummary[];
+};
+
 export const ReportView: React.FC<{
   report: LoadedReport | undefined,
 }> = ({ report }) => {
-  const searchParams = new URLSearchParams(window.location.hash.slice(1));
+  const searchParams = React.useContext(SearchParamsContext);
```

Comment:
Usually you would create a wrapper function which does it. Then you can just do

```suggestion
  const searchParams = useSearchParams();
```

Reply:
Sounds backwards to me - useContext clearly suggests that the context is used to provide the value. I don't see it recommended https://react.dev/reference/react/useContext, is it common to extract a function that makes a single call?

Reply:
It's pretty common to abstract `useContext` away when it becomes somewhat complex. As an example, lots of project have a `useAuth` hook that calls `useContext` internally to get the auth state, but that also implements some helper functions and maybe `useEffect` calls and so on.

In this specific example, I don't see a need for an abstraction. `useContext(SearchParamsContext)` and `useSearchParams()` are equal in readability to me. The moment that multiple callsites use it in similar ways though, an abstraction starts making sense :9